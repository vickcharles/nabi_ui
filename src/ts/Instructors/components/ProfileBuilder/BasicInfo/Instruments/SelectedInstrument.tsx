import * as React from 'react';
import {
  ListItem,
  ListItemSecondaryAction,
  ListItemText,
} from 'material-ui/List';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';

interface SelectedInstrumentProps {
  instrument: string;
  skillLevel: string;
  deleteInstrument: (instrumentName: string) => void;
}

const SelectedInstrument: React.StatelessComponent<SelectedInstrumentProps> = props => {
  const instrumentNameUppercase = props.instrument.charAt(0).toUpperCase() + props.instrument.slice(1);
  
  return (
    <div className="selected-instruments">
      <ListItem>
        <ListItemText
          primary={`- ${instrumentNameUppercase} (${props.skillLevel})`}
        />
        <ListItemSecondaryAction>
          <IconButton aria-label="Delete" onClick={() => props.deleteInstrument(props.instrument)}>
            <DeleteIcon />
          </IconButton>
        </ListItemSecondaryAction>
      </ListItem>
    </div>
  );
};

export default SelectedInstrument;
