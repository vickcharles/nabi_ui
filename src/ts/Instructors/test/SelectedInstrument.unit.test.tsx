import * as React from 'react';
import SelectedInstrument from '../components/BasicInfo/Instruments/SelectedInstrument';
import { shallow } from 'enzyme';

describe('SelectedInstrument', () => {
  let wrapper: any;
  const deleteInstrument: (instrumentName: string) => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <SelectedInstrument 
          instrument=""
          skillLevel=""
          deleteInstrument={deleteInstrument}
        />
      );
    });
    
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
