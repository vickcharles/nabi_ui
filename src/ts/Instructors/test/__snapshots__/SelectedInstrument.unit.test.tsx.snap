// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedInstrument Shallow Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectedInstrument
    deleteInstrument={[MockFunction]}
    instrument=""
    skillLevel=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary="-  ()"
        />
        <WithStyles(ListItemSecondaryAction)>
          <WithStyles(IconButton)
            aria-label="Delete"
            onClick={[Function]}
          >
            <pure(Delete) />
          </WithStyles(IconButton)>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>,
      "className": "selected-instruments",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(ListItemText)
            primary="-  ()"
          />,
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(IconButton)
              aria-label="Delete"
              onClick={[Function]}
            >
              <pure(Delete) />
            </WithStyles(IconButton)>
          </WithStyles(ListItemSecondaryAction)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "primary": "-  ()",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(IconButton)
              aria-label="Delete"
              onClick={[Function]}
            >
              <pure(Delete) />
            </WithStyles(IconButton)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <pure(Delete) />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(ListItem)>
          <WithStyles(ListItemText)
            primary="-  ()"
          />
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(IconButton)
              aria-label="Delete"
              onClick={[Function]}
            >
              <pure(Delete) />
            </WithStyles(IconButton)>
          </WithStyles(ListItemSecondaryAction)>
        </WithStyles(ListItem)>,
        "className": "selected-instruments",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItemText)
              primary="-  ()"
            />,
            <WithStyles(ListItemSecondaryAction)>
              <WithStyles(IconButton)
                aria-label="Delete"
                onClick={[Function]}
              >
                <pure(Delete) />
              </WithStyles(IconButton)>
            </WithStyles(ListItemSecondaryAction)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "primary": "-  ()",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(IconButton)
                aria-label="Delete"
                onClick={[Function]}
              >
                <pure(Delete) />
              </WithStyles(IconButton)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <pure(Delete) />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
